lambdas/build-stac/utils/stac.py

```diff
@@ -55,16 +61,16 @@ def generate_stac_regexevent(item: events.RegexEvent) -> pystac.Item:
     Generate STAC item from user provided regex & filename
     """
     start_datetime, end_datetime, single_datetime = regex.extract_dates(
-        item.s3_filename, item.datetime_range
+        item
     )
-
+    properties = item.properties or {}
     if start_datetime and end_datetime:
-        item.properties["start_datetime"] = f"{start_datetime.isoformat()}Z"
-        item.properties["end_datetime"] = f"{end_datetime.isoformat()}Z"
+        properties["start_datetime"] = f"{start_datetime.isoformat()}Z"
+        properties["end_datetime"] = f"{end_datetime.isoformat()}Z"
```

Comment:
There's a bug in here where if you use `dateutils.parser.parse()` on a string with a timezone (e.g. `2012-12-31T23:59:59Z`), you will get back a `datetime.datetime` object with an associated timezone:

```py
>>> parse("2012-12-31T23:59:59Z").isoformat()
'2012-12-31T23:59:59+00:00'
```

If you throw a `'Z'` on the end of that (e.g. `'2012-12-31T23:59:59+00:00Z'`), it is no longer a valid 8601 ISO formatted string and will fail validation in the STAC ingestor.